.github/workflows/ci-and-cd-pipeline.yml

```diff
@@ -0,0 +1,43 @@
+name: Continuous Integration and Continuous Delivery
+on: 
+  push:
+    paths-ignore:
+      - 'README.md'
+      - 'documentation/**'      
+  pull_request:
+    paths-ignore:
+      - 'README.md'
+      - 'documentation/**'
+jobs:
+  build:
+    name: Build and test
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - run: docker build --target test --tag todo-app:test .
+    - run: docker run --env-file ./.env.test todo-app:test
+  publish:
+    name: Publish
+    needs: build
+    runs-on: ubuntu-latest
+    if: ${{ (github.ref == 'refs/heads/main' || github.ref == 'refs/heads/module8') }}
+    steps:
+    - uses: actions/checkout@v2
+    - name: sign into Docker
+      env: 
+        DOCKER_HUB_PASSWORD: ${{secrets.DOCKER_HUB_PW}}
+      run: |
+        docker login -u anndevops123 -p $DOCKER_HUB_PASSWORD
+    - name: build prod image
+      run: docker build . --file Dockerfile --target production --tag anndevops123/ann-todo-app
+    - name: Push to Docker Hub
+      run: docker push anndevops123/ann-todo-app
+    - name: login to Heroku
```

Comment:
This name is slightly deceptive; it's logging in *and* releasing your app which is quite a major action to be hiding!